lib/cwd.js

```diff
@@ -0,0 +1,49 @@
+import {statSync} from 'node:fs';
+import {resolve} from 'node:path';
+import {fileURLToPath} from 'node:url';
+import process from 'node:process';
+
+export const getDefaultCwd = () => {
+	try {
+		return process.cwd();
+	} catch (error) {
+		error.message = `The current directory does not exist.\n${error.message}`;
+		throw error;
+	}
+};
+
+export const normalizeCwd = cwd => {
+	const cwdString = safeNormalizeFileUrl(cwd, 'The "cwd" option');
+	return resolve(cwdString);
+};
+
+export const safeNormalizeFileUrl = (file, name) => {
+	const fileString = normalizeFileUrl(file);
+
+	if (typeof fileString !== 'string') {
+		throw new TypeError(`${name} must be a string or a file URL: ${fileString}.`);
+	}
+
+	return fileString;
+};
+
+export const normalizeFileUrl = file => file instanceof URL ? fileURLToPath(file) : file;
+
+export const fixCwdError = (originalMessage, cwd) => {
+	if (cwd === getDefaultCwd()) {
+		return originalMessage;
+	}
+
+	let cwdStat;
+	try {
+		cwdStat = statSync(cwd);
```

Comment:
This function is used both by `execa()` and `execaSync()`.
In principle, we could use two separate methods, one async and one sync. This would require having two versions of `makeError()`, one async and one sync.
However, this function happens only when both:
  - the child process failed 
  - the `cwd` option was set with a different value than `process.cwd()`

So I thought it might be ok to keep it simple for the time being.